analyzers/src/SonarAnalyzer.Common/SymbolicExecution/Roslyn/OperationProcessors/CollectionTracker.cs

```diff
@@ -64,4 +64,33 @@ public static CollectionConstraint ObjectCreationConstraint(ProgramState state,
         operation.DimensionSizes.Any(x => x.ConstantValue.Value is 0)
             ? CollectionConstraint.Empty
             : CollectionConstraint.NotEmpty;
+
+    public static ProgramState LearnFrom(ProgramState state, IPropertyReferenceOperationWrapper operation, ISymbol instanceSymbol)
+    {
+        if (operation.Instance is not null
+            && operation.Property.Name is nameof(Array.Length) or nameof(List<int>.Count))
+        {
+            if (instanceSymbol is not null
+                && state[instanceSymbol]?.Constraint<CollectionConstraint>() is { } constraint)
+            {
+                var numberConstraint = constraint == CollectionConstraint.Empty
+                    ? NumberConstraint.From(0)
+                    : NumberConstraint.From(1, null);
+                state = state.SetOperationConstraint(operation, numberConstraint);
+            }
+            else if (operation.Instance.Type.DerivesOrImplementsAny(CollectionTypes))
+            {
+                state = state.SetOperationConstraint(operation, NumberConstraint.From(0, null));
+            }
+        }
+        if (operation.Property.IsIndexer)
```

Comment:
Nitpick:
```suggestion
        else if (operation.Property.IsIndexer)
```

Reply:
I'll fix it on the next PR, since I'll need to rebase